RFC-004-SWAP-Bitcoin.md

```diff
@@ -0,0 +1,124 @@
+# Bitcoin Ledger in SWAP Messages
+
+- RFC-Number: 004
+- Status: Draft
+- Discussion-issue: -
+- Created on: 1 Mar. 2019
+
+# Table of Contents
+<!-- markdown-toc start -->
+
+- [Bitcoin Ledger in SWAP Messages](#bitcoin-ledger-in-swap-messages)
+- [Table of contents](#table-of-contents)
+    - [Description](#description)
+    - [The Bitcoin Ledger](#the-bitcoin-ledger)
+        - [`network`](#network)
+    - [The Bitcoin Asset](#the-bitcoin-asset)
+        - [`quantity`](#quantity)
+    - [Registry extension](#registry-extension)
+        - [The Bitcoin Ledger](#the-bitcoin-ledger-1)
+            - [Bitcoin Networks](#bitcoin-networks)
+        - [The Bitcoin Asset](#the-bitcoin-asset-1)
+- [Examples](#examples)
+
+<!-- markdown-toc end -->
+
+## Description
+
+This RFC specifies how the Bitcoin blockchain and its native asset Bitcoin are described in RFC002 SWAP messages.
+Bitcoin refers specifically to the Bitcoin Core  [Bitcoin Core](https://github.com/bitcoin/bitcoin/) and not any blockchains derived from it.
+As required by RFC002, this RFC defines the *ledger* value `bitcoin` and its associated parameters as well as the *asset* `bitcoin`.
+This in combination with subsequent RFCs that define concrete execution steps for particular SWAP protocols enable users to execute and negotiate Bitcoin swaps.
+
+## The Bitcoin Ledger
+
+To specify Bitcoin in one of the RFC002 ledger headers (`alpha_ledger` or `beta_ledger`) use the value `bitcoin` with the following parameter:
+
+### `network`
+
+The `network` parameter describes whether you are intending to do a SWAP with *real* Bitcoin on the `mainnet` or are doing a test SWAP on the `testnet` or a shared `regtest` node.
+The `network` parameter is mandatory.
+
+All private test networks like [btcd](https://github.com/btcsuite/btcd)'s `simnet` should be specified as `regtest`.
+
+## The Bitcoin Asset
+
+Bitcoin is also the name of the Bitcoin blockchain's native asset.
+This RFC only pertains to the native Bitcoin asset.
+Other non-native assets will be defined in subsequent RFCs.
+
+Although Bitcoin is nominally the native asset, ownership of Bitcoin is really the ability to unlock *unspent transaction outputs* whose value is measured in *satoshi*.
+1 Bitcoin is simply the name for 100,000,000 satoshi.
+
+To specify Bitcoin in one of the RFC002 asset headers (`alpha_asset` or `beta_asset`) use the value `bitcoin` with the following parameter:
+
+### `quantity`
+
+The `quantity` parameter describes the quantity of satoshi that the asset represents.
+The `quantity` parameter is mandatory.
+Its value MUST be a `u64` (note that in the JSON encoding a `u64` is encoded as decimal string like `"100000000"`).
```

Comment:
🤔  I'm not an expert on this, but is `u64` only relevant for rust? What's the type in java for example? 
Maybe we should we generalize the description, e.g.:  
```suggestion
Its value MUST fit into an unsigned 64-bit integer (e.g. in rust a (u64)[https://doc.rust-lang.org/std/primitive.u64.html]) - note that in the JSON encoding a `u64` is encoded as decimal string like `"100000000"`.
```

Reply:
`u64` is what we've been calling it (see `u32` in RFC003). We don't have an RFC that defines what a `u64` or a `u32` means (we should).

Reply:
I think it's clear enough for any developer what `u64` is. Feel free to define it in the register.

Reply:
The thing we have to define is that `u64` and `u256` should always be encoded as decimal strings and that `32` should always be a JSON number.